package.json

```diff
@@ -48,13 +38,25 @@
     "generate-rcrelease": "eslint-generate-prerelease rc",
     "publish-release": "eslint-publish-release"
   },
+  "peerDependencies": {
+    "typescript": "*",
+    "eslint": ">=4.19.1",
+    "eslint-visitor-keys": ">=1.0.0"
```

Comment:
Any reason why we shouldn't also rely on the version of `eslint-visitor-keys` that's already installed as a dep of `eslint`?

Reply:
Not sure I follow. Are you asking why we don't pull in eslint-visitor-keys as a dependency, or are you saying the version here is inaccurate? (Or something else entirely?)

Reply:
Sorry, let me try to clarify: `eslint-visitor-keys` is currently a dependency. However, `eslint` already includes it as a direct dependency, leading to the possibility of there being two different versions of `eslint-visitor-keys` being used (one in `eslint` and one in `typescript-eslint-parser`). This could be mitigated by requiring `eslint-visitor-keys` as a `peerDependency` instead. I just wasn't sure if there was a reason we might actually want that behavior.

Reply:
I think it would be better to leave it as a `dependency`. Deduplicating dependencies seems like the package manager's job, not the package authors' job. To my knowledge, yarn and recent versions of npm would only install two different versions of a package if there's an incompatibility between the specified versions.

Additionally, if we made `eslint-visitor-keys` a `peerDependency`, then users would need to manually install it. (If they didn't manually install it, then their builds might unexpectedly break if `eslint` decided to change its implementation and not depend on `eslint-visitor-keys`.) So duplicate packages might end up appearing anyway if the package manager didn't deduplicate them. It seems like by using a `peerDependency` here we would be adding installation effort without much benefit.

Changing the range to `>=1.0.0` would also cause this package to break if we made breaking changes to `eslint-visitor-keys`, which seems non-ideal.

I think the only scenario where it would make sense to use a `peerDependency` here would be if we wanted the behavior of `typescript-eslint-parser` to be customizable by having the user install different versions of `eslint-visitor-keys` for it to use. But that doesn't seem to be the case here.

Reply:
That's fair! You've convinced me. I'll revert this change.